barpyrus/hlwm.py

```diff
@@ -172,10 +172,8 @@ def __init__(self,hlwm,monitor, tag_renderer = None):
         self.activecolor = hlwm('attr theme.tiling.active.color'.split(' '))
         self.emphbg = '#303030'
         self.update_tags()
-        hlwm.enhook('tag_changed', lambda a: self.update_tags(args = a))
-        hlwm.enhook('tag_flags', lambda a: self.update_tags(args = a))
-        hlwm.enhook('tag_added', lambda a: self.update_tags(args = a))
-        hlwm.enhook('tag_removed', lambda a: self.update_tags(args = a))
+        for event in ['tag_changed', 'tag_flags', 'tag_added', 'tag_removed', 'tag_renamed']:
```

Comment:
small perf note: you can use a tuple instead of a list to avoid the list allocation:
```suggestion
        for event in ('tag_changed', 'tag_flags', 'tag_added', 'tag_removed', 'tag_renamed'):
```

Reply:
What "list allocation"? A tuple also needs to be allocated, and in fact is (insignificantly) slower:

```
$ python3 -m timeit "for event in ['tag_changed', 'tag_flags', 'tag_added', 'tag_removed', 'tag_renamed']: pass"
5000000 loops, best of 5: 76.6 nsec per loop

$ python3 -m timeit "for event in ('tag_changed', 'tag_flags', 'tag_added', 'tag_removed', 'tag_renamed'): pass"
5000000 loops, best of 5: 77.3 nsec per loop
```

The golden rule when optimizing performance: Measure, don't guess :wink:

Reply:
Doh.. Well thank you, I remembered that tuples, due to their immutability and constant size, they would be always faster than lists.
But thank you for the detailed response, will use that `dis` module more often to check my assumptions! 👀

Sorry for the noise